modules/loader/source/relocations.cpp

```diff
@@ -832,6 +841,54 @@ bool resolveRISCV(const loader::Relocation &r, loader::ElfMap &map,
       writeITypeImm(getLo12(relative_value), relocation_address + 4);
       break;
     }
+    case R_RISCV_BRANCH: {
+      // R_RISCV_BRANCH immediate is for B-Type instructions and is split
+      // across 12 bits spread across the instruction and skips the bottom bit
+      // of the input relative offset. See conditional branches in the isa spec.
+
+      // Check that the value, when sign-extended back up to a 64-bit number
+      // will not lose bits.
+      if (signExtendN(relative_value, 13) !=
+          static_cast<int64_t>(relative_value & (~1))) {
+        CARGO_ASSERT(0, "R_RISCV_BRANCH branch > 13 bits or not even.");
```

Comment:
`CARGO_ASSERT` means that in debug builds, a relocation we cannot support will cause the whole program to abort, but the `return false` was in order to actually try to handle it properly by reporting a failure to the caller, right? In that case, I think we just want to print a message in debug builds and keep running? And likewise for `R_RISCV_RVC_JUMP`.

Reply:
I'm not sure it's really recoverable if we get this error tbh, since we can't do the relocation, but I'll do it if you prefer (note we do a CARGO_ASSERT - note we're not entirely consistent on this.

Reply:
We're definitely not consistent, agreed on that. If we return false, we ultimately end up at
```
  if (!loader::resolveRelocations(*elf_file, elf_map)) {
    return mux_error_internal;
  }
```
which is *almost* already what I think we would want long term: long term I think we would still want to return an error there, but have it be `mux_error_invalid_binary` which should get translated to `CL_INVALID_BINARY`. That still won't work, of course, but it's a standard error code that allows a program to report the error in a user-friendly way.